core/src/rectangle.rs

```diff
@@ -183,6 +183,17 @@ impl From<Rectangle<u32>> for Rectangle<f32> {
     }
 }
 
+impl From<Rectangle<f32>> for Rectangle<u32> {
+    fn from(rectangle: Rectangle<f32>) -> Self {
+        Rectangle {
+            x: rectangle.x as u32,
+            y: rectangle.y as u32,
+            width: rectangle.width as u32,
+            height: rectangle.height as u32,
```

Comment:
Should we `.round()` here? What's the risk of converting directly with `as`?

Reply:
There is already a `snap` method in `Rectangle` that does exactly this.

Reply:
Rounding is tricky because it may not honor the original dimensions of the rectangle (e.g. `x` may be floored but `width` may be ceiled). This can cause annoying graphical glitches.